python/cudf/cudf/_lib/io/utils.pyx

```diff
@@ -41,12 +42,20 @@ cdef source_info make_source_info(src) except*:
 
 
 # Converts the Python sink input to libcudf++ IO sink_info.
-cdef sink_info make_sink_info(src, unique_ptr[data_sink] * data) except*:
-    if isinstance(src, io.IOBase):
-        data.reset(new iobase_data_sink(src))
-        return sink_info(data.get())
-    elif isinstance(src, (int, float, complex, basestring, os.PathLike)):
-        return sink_info(<string> str(src).encode())
+cdef sink_info make_sink_info(src, unique_ptr[data_sink] * sink) except*:
+    if isinstance(src, io.TextIOBase):
+        # Files opened in text mode expect writes to be str rather than bytes,
+        # which requires conversion from utf-8. If the underlying buffer is
+        # utf-8, we can bypass this conversion by writing directly to it.
+        if codecs.lookup(src.encoding).name != "utf-8":
+            raise NotImplementedError(f"Unsupported encoding f{src.encoding}")
```

Comment:
Looks like some of the unittests are throwing:
```E   NotImplementedError: Unsupported encoding fANSI_X3.4-1968```

We might want to relax this check to also allow ascii?  
```suggestion
        if codecs.lookup(src.encoding).name not in {"utf-8", "ascii"}:
            raise NotImplementedError(f"Unsupported encoding {src.encoding}")
 ```
 (Also,  there is a minor typo in the exception - sorry if that was from me originally)

Reply:
Should that be `["utf-8", "ascii"]`?